pilot/cmd/pilot-agent/wait.go

```diff
@@ -57,6 +61,17 @@ var (
 )
 
 func checkIfReady(client *http.Client, url string) error {
+	// Check if the Envoy Proxy Server is in LIVE state
+	info, err := envoy.GetServerInfo(envoyAdminPort)
```

Comment:
Can you use this https://github.com/istio/istio/blob/master/pilot/cmd/pilot-agent/status/util/stats.go#L65 - that what is used to detect that Envoy is ready

Reply:
Why do we need any of this code anyways? It should be hitting the readiness probe which will in turn check the state for us?

Reply:
@ramaraochavali the `checkIfReady` code already calls that method. Unfortunately, the readiness stats check does not necessarily indicate traffic readiness. At least from my experience so far.